pkg/operator/internal/controllers/api-server.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+
+	operatorv1alpha1 "github.com/kubermatic/kubecarrier/pkg/apis/operator/v1alpha1"
+	apiserver "github.com/kubermatic/kubecarrier/pkg/internal/resources/api-server"
```

Comment:
Can you please add a test for it to avoid this kind of mistake that will never happen in the future? 
I had some mistake in the catalog controller and fixed in this PR: https://github.com/kubermatic/kubecarrier/pull/231,
I think for this case, a test is necessary to detect this issue in the future

Reply:
I don't think it makes sense for this PR...and furthermore as we add more features these mistakes are more obvious.

Reply:
No, I don't think the mistakes will be obvious, because actually this wrong import will:
1. miss the CRDs of api-server, which is hard to debug since you can see the CRDs are generated correctly in golden file.
2. override Tower configurations (ClusterRole, ClusterRoleBinding...) which is even harder to debug, because Tower and API server will override the same config all the time. We had the same issue of when the oeprator and manager orerrided the same secret to deploy webhooks, and it was really hard to debug.

Reply:
There are always ways to break stuff and we really need to write unittests for the operator controllers, but I don't see a need to add a test for the right import.